### PR TITLE
ARC: west runner: mdb: fix multiple argument handling issues

### LINKED ISSUE #54372: ARC: west runner: mdb: unexpected empty argument pass to MDB executable
**Describe the bug**

Currently we pass unexpected empty argument (empty quotes `''`) to MDB binary due to argument generation errors.

The example of such argument list is:

```
mdb -pset=1 -psetname=core0 '' -nooptions -nogoifmain -toggle=include_local_symbols=1 -nsim @/path/mdb_hs5x_smp.args /path/zephyr.elf
```

That happen in multiple situations:

- if target is multi-core
- if we don't specify digilent jtag device and rely on auto-select feature

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 20.04
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA: current main 156c7cd217

### LINKED ISSUE #54374: ARC: west runner: mdb: incorrect handling of unsupported jtag adapters
**Describe the bug**

Currently we don't raise error in case of the unsupported jtag adapter is selected. Thar results in calling MDB with incorrect arguments combination which won't work in any case.

**To reproduce**

- ``west build -b hsdk samples/hello_world``
- ``west flash --runner mdb-hw --jtag=ABCD``

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 20.04
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA: current main 156c7cd217

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
## Fix unexpected empty argument pass to MDB

Currently, in case of multi-core target or in case jtag adapter auto-select, we pass unexpected empty argument (empty quotes `''`) to MDB binary due to argument generation error. The example of such argument list is:
```
mdb -pset=1 -psetname=core0 '' -nooptions -nogoifmain -toggle=include_local_symbols=1 -nsim @/path/mdb_hs5x_smp.args /path/zephyr.elf
```

Such behavior looks really suspicious for anyone who check the commands produced by `west flash` or `west debug` for the `mdb-nsim` or `mdb-hw` west runners.

Fix that.

## Fix incorrect argument handling for jtag adapters

Currently we just pass in case of the unsupported jtag adapter is selected. Thar results in calling MDB with incorrect arguments combination. Fix that by rising ValueError as we do in case of other incorrect west arguments.

----

While I'm at it add few missing test-cases for mdb west runners.

----

Fixes: #54372
Fixes: #54374

### CODE REVIEW

**Review by abrodkin — DISMISSED** [review]

**Review by abrodkin — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@evgeniy-paltsev thanks -- the fixes look fine, but the release managers have requested that all pull requests which fix bugs have a bug filed before we merge them. (This is due to excessive abuse by other maintainers merging things that were features). Can you please file a bug?

**Comment by evgeniy-paltsev** [comment]
@mbolivar-nordic done!

May I ask you to re-review, thanks!

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**
